### PR TITLE
Registry output

### PR DESCRIPTION
**Pull request by amywieliczka**
This PR adds send_event_to_sns to report the success of all tasks in the harvest_collection DAG. 

It also adds an on_failure_callback to all tasks in the harvest_collection DAG.

### CODE REVIEW

**Review by barbarahui — DISMISSED** [review]

File: dags/shared_tasks/shared.py [hunk 1]
```diff
@@ -1,15 +1,72 @@
...
 
 import requests
 
 from airflow.decorators import task
 
 from urllib.parse import urlparse
 
+def send_event_to_sns(context, task_message):
+    """
+    Send a log message to a SQS FIFO queue for a specific job.
```
Comment (bibliotechy) [inline dags/shared_tasks/shared.py:13]: Is this still accurate? Should it be SNS?

**Review by bibliotechy — APPROVED** [review]
Overall looks good. A few small comments.

One overall comment: we should consider grouping the dataclasses into a single file and then importing them where used. Could help provide an overview of the data we are shipping around. Maybe there are drawbacks to this, but I like the idea of data models being together.

File: dags/shared_tasks/shared.py [hunk 2]
```diff
@@ -1,15 +1,72 @@
...
+        # TODO: this doesn't actually work to get the task parameters
+        # 'task_params': task_instance.xcom_pull(task_ids=task_instance.task_id),
+        'rikolti_message': task_message
+    }
+    message_body = json.dumps(log_message)
+
+    sns = boto3.client('sns')
+    try:
+        response = sns.publish(
+            TopicArn="arn:aws:sns:us-west-2:777968769372:RikoltiEvents",
```
[the remark this answers is not in the document]
Comment (amywieliczka) [reply dags/shared_tasks/shared.py]: I moved this into an environment variable, and also I'd been planning to lock this down in IAM permissions - only allowing the MWAA Execution Role to publish to the SNS topic and opening it up progressively and/or temporarily as needed.

**Comment by amywieliczka** [comment]
+1 for dataclass organization. I also want to investigate dataclass features a bit more closely. You can write `__init__` methods for them too. In (all?) cases in Rikolti currently, the `__init__` function I'd write is directly under the class definition. I'm going to draft up a Github issue for this and pick it off as time allows.

**Pull request merged**